planner/core/expression_rewriter.go

```diff
@@ -860,6 +860,7 @@ func (er *expressionRewriter) handleInSubquery(ctx context.Context, v *ast.Patte
 		copy(join.names, er.p.OutputNames())
 		copy(join.names[er.p.Schema().Len():], agg.OutputNames())
 		join.AttachOnConds(expression.SplitCNFItems(checkCondition))
+		join.cartesianJoin = len(join.EqualConditions) == 0
```

Comment:
Is there any test for this?

Reply:
In Subquery.test `explain select * from t1 where t1.a in (select sum(b) from t2);`